stonesoup/types/tests/test_association.py

```diff
@@ -89,7 +95,9 @@ def test_associationset():
                                    timestamp=timestamp1)
 
     assoc2 = TimeRangeAssociation(objects=set(objects_list[1:]),
-                                  time_range=timerange)
+                                  time_range=time_range)
+    assoc_duplicate = TimeRangeAssociation(objects=set(objects_list[1:]),
+                                           time_range=time_range2)
```

Comment:
I could have been clearer - it's a duplicate because the objects are the same. This should lead to it becoming merged together with `assoc2` by the `AssociationSet._simplify` method when used in the same `AssociationSet`. I'll rename a few variables to make it clearer